federationtypes.go

```diff
@@ -0,0 +1,97 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// A SendRequest is the content of a request to PUT /_matrix/federation/v1/send/{txnID}/
+type SendRequest Transaction
```

Comment:
Alternatively, if we want the warm fuzzies of typing requests/responses, I would advocate using `ReqXXX` and `RespXXXX` to make it clearer that:
 - They don't look like callable functions at call-sites.
 - They sort nicely on godoc.